python/tvm/script/parser.py

```diff
@@ -440,8 +463,23 @@ def check_decorator(decorators: List[ast.Expr]) -> bool:
         # add parameters of function
         for arg in node.params:
             arg_var = tvm.te.var(arg.name, self.parse_type(arg.ty, arg))
```

Comment:
since this is not used in match buffer case, move it to `else` branch

Reply:
This is actually used in match buffer syntax sugar case as the `arg.name` needs to later fetched from `self.context.func_params` to be part of match_buffer's argument.